SignalServiceKit/src/Devices/OWSLinkedDeviceReadReceipt.m

```diff
@@ -0,0 +1,53 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSLinkedDeviceReadReceipt.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation OWSLinkedDeviceReadReceipt
+
+- (instancetype)initWithSenderId:(NSString *)senderId timestamp:(uint64_t)timestamp;
+{
+    OWSAssert(senderId.length > 0 && timestamp > 0);
+
+    self = [super initWithUniqueId:[OWSLinkedDeviceReadReceipt uniqueIdForSenderId:senderId timestamp:timestamp]];
+    if (!self) {
+        return self;
+    }
+
+    _senderId = senderId;
+    _timestamp = timestamp;
+
+    return self;
+}
+
++ (NSString *)uniqueIdForSenderId:(NSString *)senderId timestamp:(uint64_t)timestamp
+{
+    OWSAssert(senderId.length > 0 && timestamp > 0);
+
+    return [NSString stringWithFormat:@"%@ %llu", senderId, timestamp];
```

Comment:
Super nit: can we use a dash or underscore rather than a space in the id. An identifier containing whitespace gives me a real case of the phantods.

Reply:
![as you wish](https://user-images.githubusercontent.com/625803/30927776-922e1d22-a387-11e7-9b09-c3356fadef0a.gif)